internal/server/server.go

```diff
@@ -52,10 +52,12 @@ type Server struct {
 	toolsets     map[string]tools.Toolset
 }
 
-func InitializeConfigs(ctx context.Context, cfg ServerConfig, l log.Logger, instrumentation *Instrumentation) (map[string]sources.Source, map[string]auth.AuthService, map[string]tools.Tool, map[string]tools.Toolset, error) {
+func InitializeConfigs(ctx context.Context, VersionIn string, SourceConfigsIn SourceConfigs, AuthServiceConfigsIn AuthServiceConfigs, ToolConfigsIn ToolConfigs, ToolsetConfigsIn ToolsetConfigs, l log.Logger, instrumentation *Instrumentation) (map[string]sources.Source, map[string]auth.AuthService, map[string]tools.Tool, map[string]tools.Toolset, error) {
```

Comment:
nit: this line is past readable -- suggest splitting it up over a few lines. 

It's probably also time to use a struct to pass them in as well.